src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -175,7 +175,8 @@ int MacroAssembler::patch_oop(address insn_addr, address o) {
   // instruction.
   if (Instruction_aarch64::extract(insn, 31, 21) == 0b11010010101) {
     // Move narrow OOP
-    narrowOop n = CompressedOops::encode((oop)o);
+    narrowOop noop = CompressedOops::encode((oop)o);
+    CompressedOops::NarrowType n = CompressedOops::narrow_oop_value(noop);
```

Comment:
I think a public NarrowType is confusing (vs narrowOop). Since NarrowType is rarely used, and it's unlikely to be changed from uint32_t, maybe we should make NarrowType private, and change the above to
```
uint32_t n = CompressedOops::encode_as_uint32_t((oop)o);
```
Code that does arithmetic operations on the integer value probably wants to know the exact type anyway. I.e., not knowing whether NarrowType is signed or not makes me worry about the correctness of `n >> 16` below.